tfhe/src/core_crypto/algorithms/misc.rs

```diff
@@ -108,6 +108,25 @@ pub fn torus_modular_diff<T: UnsignedInteger>(
     }
 }
 
+// Our representation of non native power of 2 moduli puts the information in the MSBs and leaves
+// the LSBs empty, this is what this function is checking
+pub fn check_content_respects_mod<Scalar: UnsignedInteger, Input: AsRef<[Scalar]>>(
+    input: &Input,
+    modulus: CiphertextModulus<Scalar>,
+) -> bool {
+    if !modulus.is_native_modulus() {
```

Comment:
I just realized this function only checks power of 2 moduli

the code needs to be slightly adapted (my bad) but this is to avoid unwanted panics/asserts

so you will need 

```rust
if modulus.is_native_modulus() {
    true
} else if modulus.is_power_of_two() {
    // the existing code
} else {
   // non native power of two 
   let scalar_modulus: Scalar = modulus.get_custom_modulus().cast_into();
   return input
            .as_ref()
            .iter()
            .all(|&x| x < scalar_modulus);
}
```